packages/blaze-tx/src/utils.ts

```diff
@@ -46,3 +55,64 @@ export function calculateReferenceScriptFee(
 
   return refFee;
 }
+
+export interface IScriptData {
+  redeemersEncoded: string;
+  datumsEncoded: string | undefined;
+  costModelsEncoded: string;
+  hashedData: HexBlob;
+  scriptDataHash: Hash32ByteBase16;
+}
+
+/**
+ * Calculates the correct script data hash for a transaction
+ *
+ * Separate from the `getScriptData` method in `TxBuilder` to allow for more thorough testing
+ * This is heavily documented here:
+ * https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl#L423-L490
+ *
+ * @param redeemers - The redeemers of the transaction
+ * @param datums - The datums in the witness set of the transaction
+ * @param usedCostModels - The cost models for any languages used in the transaction
+ */
+export function computeScriptDataHash(
```

Comment:
If we're making this a utility function, we can probably skip wrapping the private method from `TxBuilder` and just write tests for this. Also, since it's more than the script data hash, why not rename to `computeScriptData`?

Reply:
the rename is probably good; I pulled this out so we *could* write tests directly for this, but then decided against it, since constructing the fields to pass in is just as tedious, and I want to test that the transaction pulls the right fields in the first place.